lightning/src/ln/channelmanager.rs

```diff
@@ -1098,13 +1102,20 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref> ChannelMan
 						break Some(("Forwarding node has tampered with the intended HTLC values or origin node has an obsolete cltv_expiry_delta", 0x1000 | 13, Some(self.get_channel_update(chan).unwrap())));
 					}
 					let cur_height = self.latest_block_height.load(Ordering::Acquire) as u32 + 1;
-					// We want to have at least LATENCY_GRACE_PERIOD_BLOCKS to fail prior to going on chain CLAIM_BUFFER blocks before expiration
+					// Theoretically, channel counterparty shouldn't send us a HTLC expiring now, but we want to be robust wrt to counterparty
+					// packet sanitization.
 					if msg.cltv_expiry <= cur_height + CLTV_CLAIM_BUFFER + LATENCY_GRACE_PERIOD_BLOCKS as u32 { // expiry_too_soon
 						break Some(("CLTV expiry is too close", 0x1000 | 14, Some(self.get_channel_update(chan).unwrap())));
 					}
 					if msg.cltv_expiry > cur_height + CLTV_FAR_FAR_AWAY as u32 { // expiry_too_far
 						break Some(("CLTV expiry is too far in the future", 21, None));
 					}
+					// In theory, we would be safe against unitentional channel-closure, if we only required a margin of LATENCY_GRACE_PERIOD_BLOCKS.
+					// But, to be safe against policy reception, we use a longuer delay.
+					if (*outgoing_cltv_value) as u64 <= (cur_height + 2 * CLTV_CLAIM_BUFFER + LATENCY_GRACE_PERIOD_BLOCKS) as u64 {
```

Comment:
Why 2* CLAIM_BUFFER? We never need two on-chain claim buffers? I could understand an argument for 2*LATENCY_GRACE_PERIOD, because if we presume that LATENCY_GRACE_PERIOD it the max time it should take to do a commitment_signed dance then you could suggest we need two, but that's not quite true. We fail-back after the expiry minus LATENCY_GRACE_PERIOD, not even GRACE_PERIOD + CLAIM_BUFFER.

Reply:
Ultimately LATENCY_GRACE_PERIOD + CLTV_CLAIM_BUFFER seems like more than enough - for outbound stuff we don't really care, its up to our counterparty to care about the height here, we just don't want to get force-closed on because we're behind by one block and are too tight.

Reply:
2*CLAIM_BUFFER is increasing how long someone would have to mess up with our onchain view before to somehow steal HTLC. But this has other implication on outbound policy of other implementation I guess so reverse for now.

Reply:
And in fact CLAIM_BUFFER is 6, so we have 9 blocks which is not so bad.